build/configs/toolchain_linux_x86_64.cmake

```diff
@@ -15,9 +15,15 @@
 set(CMAKE_SYSTEM_NAME Linux)
 set(CMAKE_SYSTEM_PROCESSOR x86_64)
 
-find_program(CMAKE_C_COMPILER NAMES x86_64-linux-gnu-gcc x86_64-unknown-linux-gnu-gcc)
-find_program(CMAKE_CXX_COMPILER NAMES x86_64-linux-gnu-g++ x86_64-unknown-linux-gnu-g++)
+if(NOT DEFINED CMAKE_C_COMPILER)
+  find_program(CMAKE_C_COMPILER NAMES x86_64-linux-gnu-gcc x86_64-unknown-linux-gnu-gcc)
```

Comment:
@akiss77, maybe, we should introduce some set of `toolchain_*.cmake` files for different compilers. The files would set compiler executable names and also apply compiler-specific values of build options (for example `FLAGS_LTO=-flto` for gcc / g++) / warnings configuration etc.
Also, maybe we should extract the architecture-specific part to sub-configuration files for each supported compiler-architecture pair and put compiler-architecture specific flags / options to the files.

In the case, we could remove gcc / g++ specific `if` blocks from CMakeLists.txt, and also would support clang build for arhitectures other than x86_64.

What do you think about this?